### ISSUE TITLE
getDocumentHighlights does not return uses of SFCs in JSX elements

### ISSUE THREAD

**Issue by myitcv** [Microsoft/TypeScript#7109]
This issue first surfaced in https://github.com/palantir/tslint/issues/972

I think the following is a minimal reproduction of the problem:

```typescript
/// <reference path='fourslash.ts' />

//@Filename: file.tsx
//// export function BuildMyComponent(): JSX.Element {
//// 	let [|StringComponentA|]: React.ComponentClass<{}>;
//// 	return <[|StringCom/**/ponentA|] />;
//// }
```

The issue first surfaced in `1.9.0-dev.20160212`; indeed I think I've bisected the problem to 13167f53e6160bb611c416a3c40925ff4ab8441a using the above test

**Comment by myitcv** [comment]
@mhegazy @RyanCavanaugh - anyone using `tslint` with the `no-unused-variable` rule (and React) is currently blocked from upgrading past `1.9.0-dev.20160212` (details https://github.com/palantir/tslint/issues/972) because of this issue.

Is there any chance therefore that this could be addressed before the `2.0` milestone?

That would then allow us to pick up and test things like [private constructors](https://github.com/Microsoft/TypeScript/pull/6885) which landed in more recent nightlies. 

Thanks in advance

**Comment by RyanCavanaugh** [comment]
We'd take a PR at any time but I have other more urgent work at the moment. It'll probably be a week or two before I get to this.

**Comment by myitcv** [comment]
Thanks Ryan

**Comment by myitcv** [comment]
Confirmed this is fixed by #7622 (thanks @mhegazy) available as of `1.9.0-dev.20160322`
